### PR TITLE
feat(discover): add timestamp.to_{hour,day}

### PR DESCRIPTION
**Pull request by alxu-sentry**
The "time" field was removed in #24136 due to being confusing and poorly
defined. However, the concept of grouping events by hour is still valuable.
Therefore, add a new field to support this concept. Grouping by day is also
reasonable and can easily be added at the same time, so do it here. Grouping by
month is also plausible, but Sentry currently only stores data up to 90 days,
so showing only three months (two of which are truncated) is likely of limited
value.

This is exposed to the user as a field instead of a function because for
backend implementation simplicity, it does not aggregate results (i.e. there
will likely be semi-identical output rows). Users will generally want to add an
aggregation function, such as count(), when using these fields.

timestamp.to_day is currently displayed as a full date and time, always at
00:00 UTC. This is not great, but it's not that easy to fix it cleanly and I
think this is likely to be infrequently used anyways.

Screen Shot 2021-03-11 at 5 38 03 PM

### CODE REVIEW

File: tests/sentry/snuba/test_discover.py [hunk 1]
```diff
@@ -132,40 +133,27 @@ def test_missing_project(self):
...
         assert data[0]["release"] == "first-release"
 
-        assert len(result["meta"]) == 3
-        assert result["meta"] == {
-            "project.id": "integer",
-            "user": "string",
-            "release": "string",
-        }
-
-    def test_field_alias_with_component(self):
```
Comment (alxu-sentry) [inline tests/sentry/snuba/test_discover.py:152]: hm, actually more investigation is required before removing this test.

Comment (alxu-sentry) [reply tests/sentry/snuba/test_discover.py:152]: going to move the removal of this test to another PR.

Comment (dashed) [reply tests/sentry/snuba/test_discover.py:152]: What's the reason for removing this test? This was added in https://github.com/getsentry/sentry/pull/18192

Comment (alxu-sentry) [reply tests/sentry/snuba/test_discover.py:152]: Yes, I saw. I unremoved it in https://github.com/getsentry/sentry/pull/24387/commits/70232534cfec3865ae72cc362044de00254f1e25.

Comment (dashed) [reply tests/sentry/snuba/test_discover.py:152]: Preferably this test shouldn't be removed as we like to ensure the user fields are consistent with the behaviour we expect. The user fields had a history of causing confusion to the end user.

**Review by dashed — APPROVED** [review]

**Pull request merged**